src/wallet/walletdb.cpp

```diff
@@ -726,7 +732,7 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
             {
                 // losing keys is considered a catastrophic error, anything else
                 // we assume the user can live with:
-                if (IsKeyType(strType) || strType == DBKeys::DEFAULTKEY) {
+                if (wss.corrupt || IsKeyType(strType) || strType == DBKeys::DEFAULTKEY) {
                     result = DBErrors::CORRUPT;
                 } else if (strType == DBKeys::FLAGS) {
                     // reading the wallet flags can only fail if unknown flags are present
```

Comment:
should the other corruption reasons also latch to true?

`wss.corrupt` would never be set to false, so the error `TOO_NEW` is translated to the error `CORRUPT` (after at least one CORRUPT error).

With a lost key this doesn't seem to happen and the error might be rewritten to TOO_NEW.

Probably doesn't matter in practice, but I wanted to ask to be sure.